checkov/main.py

```diff
@@ -572,349 +580,361 @@ def run(self, banner: str = checkov_banner) -> int | None:
-            bc_integration.persist_scan_results(self.scan_reports)
-            bc_integration.persist_image_scan_results(runner.raw_report, self.config.dockerfile_path, self.config.docker_image,
-                                                      self.config.branch)
-            bc_integration.persist_run_metadata(self.run_metadata)
-            self.url = self.commit_repository()
-            exit_code = self.print_results(runner_registry=runner_registry, url=self.url)
-            return exit_code
-        elif self.config.file:
-            runner_registry.filter_runners_for_files(self.config.file)
-            self.scan_reports = runner_registry.run(
-                external_checks_dir=external_checks_dir,
-                files=self.config.file,
-                repo_root_for_plan_enrichment=self.config.repo_root_for_plan_enrichment,
-            )
-            if runner_registry.is_error_in_reports(self.scan_reports):
-                self.exit_run()
-            if baseline:
-                baseline.compare_and_reduce_reports(self.scan_reports)
-            if self.config.create_baseline:
-                overall_baseline = Baseline()
-                for report in self.scan_reports:
-                    overall_baseline.add_findings_from_report(report)
-                created_baseline_path = os.path.join(os.path.abspath(os.path.commonprefix(self.config.file)),
-                                                     '.checkov.baseline')
-                with open(created_baseline_path, 'w') as f:
-                    json.dump(overall_baseline.to_dict(), f, indent=4)
-
-            if bc_integration.is_integration_configured():
-                files = [os.path.abspath(file) for file in self.config.file]
-                root_folder = os.path.split(os.path.commonprefix(files))[0]
-
-                self.upload_results(
-                    root_folder=root_folder,
-                    files=files,
-                    excluded_paths=runner_filter.excluded_paths,
-                    git_configuration_folders=git_configuration_folders,
                 )
-            exit_code = self.print_results(
-                runner_registry=runner_registry,
-                url=self.url,
-                created_baseline_path=created_baseline_path,
-                baseline=baseline,
-            )
-            return exit_code
-        elif not self.config.quiet:
-            print(f"{banner}")
+                return exit_code
+            elif not self.config.quiet:
+                print(f"{banner}")
 
-            bc_integration.onboarding()
-        return None
+                bc_integration.onboarding()
+            return None
+        finally:
+            if self.config.support:
+                if not self.config.bc_api_key:
+                    self.parser.error("--bc-api-key argument is required when using --support")
+                else:
+                    bc_integration.persist_run_metadata(log_stream.getvalue())
```

Comment:
@reemodo So we invoke a function called "persist_**run_metdata**" in order to persist **logs**?
Perhaps we should create another function, or change the name of the current one?